### PR TITLE
@astrojs/tailwind: simplify, upgrade & fix support for ts config file

### PR DESCRIPTION
**Pull request by TomPichaud**
First participation to Astro! Love your project and looking forward to its evolution!

Fix: https://github.com/withastro/astro/issues/6695

## Changes

Commit https://github.com/withastro/astro/pull/6724/commits/8a8be4221edb136af505d17f0f20ec1d341f6a1a : add support for ESM/TS config files while maintaining current behaviour: providing a default tailwind config for astro if none is present.

Knowing that Astro can already suggest and generate a default tailwind config trough the command `astro add tailwind`, I think we can simplify this integration further, reducing its dependencies and making it more robust.

In light of the above, I've made the following commit https://github.com/withastro/astro/pull/6724/commits/4a515e3dc402a705c8ffcbe35a839904c2f6e190 : only setup postcss, supporting custom config file path and the base css import

Also, I feel like any integration setup should be inside the integration package and not hard coded into astro to be able to use `astro add ...`.

## Testing

Tested locally with a `tailwind.config.ts` and custom `custom-tailwind.config.ts`

## Docs

No further documentation is needed, but we could add ESM/TS support

### CODE REVIEW

**Comment by TomPichaud** [comment]
Commit https://github.com/withastro/astro/pull/6724/commits/8a8be4221edb136af505d17f0f20ec1d341f6a1a add support for ESM/TS config files while maintaining current behaviour: providing a default tailwind config file for astro

Commit https://github.com/withastro/astro/pull/6724/commits/4a515e3dc402a705c8ffcbe35a839904c2f6e190 simplify this integration by only setting up postcss, supporting custom config file path and the base css import

File: .changeset/real-spies-pretend.md
Comment (matthewp) [inline .changeset/real-spies-pretend.md:2]: Mistake?

**Comment by matthewp** [comment]
Hey @TomPichaud are you still working on this change? Thanks!

**Comment by TomPichaud** [comment]
I think I'm done. Did you read all of my commit ? I was hoping for some review or discussion around this but I'm happy with my work. Maybe someone can review and approve ?

I'm not sure about the version as I've made some changes that affect users with no tailwind config, if any ¿? But like I said, Astro is already pushing the user to have a tailwind config file and can do it through the CLI.

**Comment by chav-aniket** [comment]
+1 for getting this reviewed asap 🙇🏽

**Comment by TomPichaud** [comment]
@matthewp Any review would be welcome please ? It's working perfectly on all my projects.

**Comment by helbing** [comment]
@TomPichaud Hi guy, are you still dealing with this PR?

**Comment by TomPichaud** [comment]
Just waiting for someone to review and approve...

**Comment by matthewp** [comment]
We usually don't review the code until the PR is ready; meaning merge conflicts are resolved and tests are passing. 

As for the overall theme of the PR, definitely in favor of simplifying! One thing that's important is HMR. Have you verified that this change still allows Tailwind changes to HMR in the client? Have you verified that changing the config triggers HMR as well (the `addWatchFile` removal might have hurt that).

And just to clarify, the main new feature / bug fix of this PR is that TypeScript and ESM configs are now supported, is that correct?

**Comment by helbing** [comment]
I see this PR want to reactor the Tailwind integration and then support the TypeScript config to fix errors. I think we should separate fix things and refactor things. Support the TypeScript config is high priority. We should deal with it first and do refactor later in other PR. How do you think? @TomPichaud @matthewp

**Comment by helbing** [comment]
I think we should separate the fix thing and refactor thing. To support TypeScript config is high priority, we should deal with it first. And then later we can refactor the tailwind integration at other PR. How do you think? @TomPichaud @matthewp

**Comment by TomPichaud** [comment]
Like I said, there's little point doing redundant work. To my opinion it's best to let tailwind load the config and not do it ourself, adding more dependencies, bug possibilities and the necessity to update it when tailwind add a feature. Simple|Less is better, and if we follow tailwind's recommendations we just need to configure postcss + vite, like I'm suggesting we do.

Also they are working on automatic 'content' detection so it will already be ready for that.

So I disagree on the fix first and then refactor because the fix is adding dependencies as the refactoring is removing some dependencies. To get the same result.
Also Astro'doc already recommend and explain how to create a tailwind config file that support astro, and it even do it for you through the command line using tailwind init feature. So we should not overthink it and load a custom config that we edit just for the content section to support astro.

**Comment by helbing** [comment]
> Like I said, there's little point doing redundant work. To my opinion it's best to let tailwind load the config and not do it ourself, adding more dependencies, bug possibilities and the necessity to update it when tailwind add a feature. Simple|Less is better, and if we follow tailwind's recommendations we just need to configure postcss + vite, like I'm suggesting we do.
> 
> Also they are working on automatic 'content' detection so it will already be ready for that.
> 
> So I disagree on the fix first and then refactor because the fix is adding dependencies as the refactoring is removing some dependencies to get the same result. Also Astro'doc already recommend and explain how to create a tailwind config file that support astro, and it even do it for you through the command line using tailwind init feature. So we should not overthink it and load a custom config that we edit just for the content section to support astro.
> 
> I suggest you test this pr thoroughly and merge it. I've got no issues locally.

Yes, I test it locally, it can works. I see the branch has conflicts need be resolved, can you deal with it?

**Comment by bluwy** [comment]
I've rebased the PR and it should be good to go now. I also think it's nice to clean up the config by moving `config.path` and `config.applyBaseStyles` to the root, which that could be made after this is merged.

**Comment by TomPichaud** [comment]
Thanks for your help!

File: .changeset/real-spies-pretend.md
Comment (ematipico) [reply .changeset/real-spies-pretend.md:2]: I can't see the history of the file. @TomPichaud is this change **intentional**? Shouldn't this be a `patch` change, considering the PR says  "fix"? 

If this is a `major`, and breaks existing APIs, the changeset should mention a migration path for what's changed.

**Comment by matthewp** [comment]
Let's do a preview release and test it out. In particular we need to verify there are no regressions in terms of HMR, which is something we fixed in the previous implementation.

**Comment by matthewp** [comment]
!preview tailwind-config

**Comment by matthewp** [comment]
For some reason the preview release didn't work. So I manually did a beta release which you can test with:

```shell
npm install @astrojs/tailwind@beta
```

**Comment by matthewp** [comment]
HMR looks good here.

**Review by ematipico — APPROVED** [review]
Let's 🚢 it!

File: .changeset/real-spies-pretend.md
Comment (bluwy) [reply .changeset/real-spies-pretend.md:2]: I've updated this part with what changed and the migration path.

**Pull request merged**
